sdk/tests/iot/pnp/test_az_iot_pnp_client_twin.c

```diff
@@ -0,0 +1,1092 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "test_az_iot_pnp_client.h"
```

Comment:
There is a lot of test code here. I am going to skip reviewing these, but I am sure they are great!

If someone wants to do a spot-check, please do, but as long as coverage is good and the implementation/API is correct, I think we have our bases covered :)